pkg/buildah/imageCrdBuilder.go

```diff
@@ -0,0 +1,310 @@
+// limitations under the License.
+
+package buildah
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/openconfig/gnmi/errlist"
+
+	"github.com/containers/buildah"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	imagev1 "github.com/labring/sealos/controllers/imagehub/api/v1"
+	"github.com/labring/sealos/pkg/client-go/kubernetes"
+	"github.com/labring/sealos/pkg/utils/file"
+	"github.com/labring/sealos/pkg/utils/logger"
+)
+
+const (
+	AppPath          = "app"
+	READMEpath       = "README.md"
+	ConfigFileName   = "config.yaml"
+	AuthjsonPath     = "containers/auth.json"
+	SealosPath       = ".sealos"
+	ImagehubGroup    = "imagehub.sealos.io"
+	ImagehubVersion  = "v1"
+	ImagehubResource = "images"
+	ImagehubKind     = "Image"
+	XdgRuntimeDir    = "XDG_RUNTIME_DIR"
+)
+
+const (
+	//Read Content Output
+	ReadOrBuildAppConfigOutput = "Success get appConfig"
+	ReadImageContentOutput     = "Success read image content"
+	ReadReadmeOutput           = "Success read readme file"
+)
+
+type ImageCRDBuilder struct {
+	imagename      string
+	repositoryname string
+	clustername    string
+	username       string
+	userconfig     string
+	AppConfig      *imagev1.Image
+}
+
+type dockerAuthConfig struct {
+	Auth          string `json:"auth,omitempty"`
+	IdentityToken string `json:"identitytoken,omitempty"`
+}
+
+type dockerConfigFile struct {
+	AuthConfigs map[string]dockerAuthConfig `json:"auths"`
+	CredHelpers map[string]string           `json:"credHelpers,omitempty"`
+}
+
+func NewAndRunImageCRDBuilder(args []string) {
+	//run without error returns
+	icb := &ImageCRDBuilder{args[0], strings.Split(args[0], "/")[0], "", "", "", &imagev1.Image{}}
+	if icb.CheckLoginStatus() {
+		logger.Info("start ImageCrd Push")
+		icb.Run()
+	}
+}
+
+func (icb *ImageCRDBuilder) CheckLoginStatus() bool {
+	//Check Cri Login
+	var path string
+	var dockerConfig dockerConfigFile
+	if path = os.Getenv(XdgRuntimeDir); path == "" {
+		return false
+	}
+	if !file.IsExist(filepath.Join(path, AuthjsonPath)) {
+		return false
+	}
+	encodeAuth, err := file.ReadAll(filepath.Join(path, AuthjsonPath))
+	if err != nil {
+		return false
+	}
+	err = json.Unmarshal(encodeAuth, &dockerConfig)
+	if err != nil {
+		return false
+	}
+	//auth struct is username:password which is namespcae:kubeconfig in sealos.hub scenario
+	auth, err := base64.StdEncoding.DecodeString(dockerConfig.AuthConfigs[icb.repositoryname].Auth)
+	if err != nil {
+		return false
+	}
+	icb.username = strings.SplitN(string(auth), ":", 2)[0]
```

Comment:
same as above?

Reply:
ok

Reply:
I double check this code ,It different from above .It read auth.json ,the struct in this file is different with simple docker domin parse,so this code I think better remain.